dvc/command/plot.py

```diff
@@ -0,0 +1,239 @@
+            revisions.append(WORKSPACE_REVISION_NAME)
+        return revisions
+
+
+def add_parser(subparsers, parent_parser):
+    PLOT_HELP = (
+        "Generating plots for metrics stored in structured files "
+        "(json, csv, tsv)."
+    )
+
+    plot_parser = subparsers.add_parser(
+        "plot",
+        parents=[parent_parser],
+        description=append_doc_link(PLOT_HELP, "plot"),
+        help=PLOT_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    plot_subparsers = plot_parser.add_subparsers(
+        dest="cmd",
+        help="Use `dvc plot CMD --help` to display command-specific help.",
+    )
+
+    fix_subparsers(plot_subparsers)
+
+    SHOW_HELP = "Plot data from a file."
+    plot_show_parser = plot_subparsers.add_parser(
+        "show",
+        parents=[parent_parser],
+        description=append_doc_link(SHOW_HELP, "plot/show"),
+        help=SHOW_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    plot_show_parser.add_argument(
+        "-t",
+        "--template",
+        nargs="?",
+        default=None,
+        help="File to be injected with data.",
+    )
+    plot_show_parser.add_argument(
+        "datafile", nargs="?", default=None, help="Data to be visualized."
+    )
+    plot_show_parser.add_argument(
+        "-r", "--result", help="Name of the generated file."
+    )
+    plot_show_parser.add_argument(
+        "--no-html",
+        action="store_true",
+        default=False,
+        help="Do not wrap vega plot json with HTML.",
+    )
+    plot_show_parser.add_argument(
+        "-f",
+        "--fields",
+        default=None,
+        help="Choose which fileds or jsonpath to put into plot.",
+    )
+    plot_show_parser.add_argument(
+        "-x", default=None, help="Field name that will be on x axis of plot."
+    )
+    plot_show_parser.add_argument(
+        "-y", default=None, help="Field name that will be on y axis of plot."
+    )
+    plot_show_parser.add_argument(
+        "-o",
+        "--stdout",
+        action="store_true",
+        default=False,
+        help="Print result to stdout.",
+    )
+    plot_show_parser.add_argument(
+        "--no-csv-header",
+        action="store_true",
+        default=False,
+        help="Provided CSV ot TSV datafile does not have a header.",
+    )
+    plot_show_parser.set_defaults(func=CmdPlotShow)
+
+    PLOT_DIFF_HELP = (
+        "Plot changes between commits in the DVC repository, "
+        "or between the last commit and the workspace."
+    )
+    plot_diff_parser = plot_subparsers.add_parser(
+        "diff",
+        parents=[parent_parser],
+        description=append_doc_link(PLOT_DIFF_HELP, "plot/diff"),
+        help=PLOT_DIFF_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    plot_diff_parser.add_argument(
+        "-t",
+        "--template",
+        nargs="?",
+        default=None,
+        help=("File to be injected with data."),
```

Comment:
Template is Vega json specificaiton without data.
example:
https://vega.github.io/editor/#/examples/vega-lite/point_2d

Reply:
Got it. I need to study Vega graph basics...

Reply:
Is there a link to Vega template documentaion? I found https://vega.github.io/vega/docs/specification/ but that's about the format, not really about template files.

Or are templates a DVC feature instead? And if so where are we explaining them??? Thanks 🙂

Reply:
The templates are our thing. They can contain some DVC-specific content that can be only interpreted by DVC. I think we should link to the documentation when it will be available.

Reply:
Is there an issue in the dvc.org repo to document templates? I just don't want to forget about this. Or maybe just a checkbox inside a general issue about documenting plots...

Reply:
ok I added a checkbox in https://github.com/iterative/dvc.org/pull/1186#pullrequestreview-407626236